sources/xfscli/src/bin/fsck_xfs.rs

```diff
@@ -0,0 +1,199 @@
+    let force = args.force;
+    let repair = args.repair;
+
+    // This is diverted behavior from xfs shell script.
+    // In fsck no filesystems are specified on the command line,
+    // and the -A option is not specified,
+    // fsck will default to checking filesystems in /etc/fstab serially
+    // Hence device name is not a mandatory argument. But as there isn't a /etc/fstab
+    // file in Bottlerocket we have to specify the device.
+    let target = &args.target.as_ref().context(error::ParseTargetSnafu)?;
+
+    if !Path::new(target).exists() {
+        return Err(Error::DeviceNotFound {
+            target: target.to_string(),
+        });
+    }
+
+    if force {
+        // Preform xfs_repair
+        println!("Performing XFS repair..");
+        let mut repair_exit_code = perform_xfs_repair(vec!["-e", target]);
+
+        // Clear the log, mount and unmount the XFS file system
+        // before xfs_repair
+        if repair_exit_code == XfsRepairResponseCode::DirtyLogs && repair {
+            println!("Replaying log for {}", target);
+
+            // Using random directory (/tmp/repair_mntxxxx) instead of a targeted directory (/tmp/repair_mnt)
+            let mnt_dir = tempfile::Builder::new()
+                .prefix(REPAIR_MOUNT_DIR)
+                .tempdir()
+                .context(error::MakeDirSnafu)?;
+
+            let mnt_dir_path = mnt_dir
+                .path()
+                .to_str()
+                .expect("Unable to get path of temporary dir");
+
+            let kernel_params = get_root_and_rootflags(target);
+
+            let basename_dev = Path::new(target)
+                .file_name()
+                .expect("Unable to find device base name");
+
+            let basename_root = Path::new(&kernel_params.root)
+                .file_name()
+                .expect("Unable to find root base name");
+
+            let mut mount_args = vec![target.as_str(), mnt_dir_path];
+            if basename_dev == basename_root && !kernel_params.root_flags.is_empty() {
+                mount_args.push(kernel_params.root_flags.as_str());
+            }
+
+            let mount_status = Command::new(MOUNT)
+                .args(mount_args)
+                .status()
+                .expect("Couldn't run mount");
+
+            if !mount_status.success() {
+                return Err(Error::Mount);
+            }
+
+            Command::new(UMOUNT)
+                .arg(mnt_dir_path)
+                .output()
+                .expect("Couldn't run unmount");
+
+            println!("Performing XFS repair again after cleaning the log..");
+            repair_exit_code = perform_xfs_repair(vec!["-e", target]);
+
+            let mnt_dir_close_msg = format!("Could not delete the {} dir", mnt_dir_path);
+            mnt_dir.close().expect(&mnt_dir_close_msg);
+        }
+
+        let metadata_repaired = match repair_exit_code {
+            XfsRepairResponseCode::Ok => Ok(false),
```

Comment:
I don't know the details of the underlying software, but I'm a bit surprised that `Ok` translates to `false`.

Reply:
This function return success in 2 cases after running xfs_repair. 
1. When filesysten itself is repaired 
2. When metadata has been repaired.

This match is to understand if the success is because of metadata repair or filesystem repair. Then later `if metadata_repaired { 1 } else { 0 }` this returns the exact result with the expected code.